pygmt/datasets/samples.py

```diff
@@ -95,51 +95,31 @@ def load_sample_data(name):
     return data
 
 
-def load_japan_quakes(**kwargs):
+def _load_japan_quakes():
     """
-    (Deprecated) Load a table of earthquakes around Japan as a
-    pandas.DataFrame.
-
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="japan_quakes")`` and will be removed in
-       v0.9.0.
-
-    Data is from the NOAA NGDC database. This is the ``@tut_quakes.ngdc``
-    dataset used in the GMT tutorials.
-
-    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
-    first time you invoke this function. Afterwards, it will load the data from
-    the cache. So you'll need an internet connection the first time around.
+    Load a table of earthquakes around Japan as a pandas.DataFrame.
 
     Returns
     -------
     data : pandas.DataFrame
         The data table. Columns are year, month, day, latitude, longitude,
         depth (in km), and magnitude of the earthquakes.
     """
-
-    if "suppress_warning" not in kwargs:
-        warnings.warn(
-            "This function has been deprecated since v0.6.0 and will be "
-            "removed in v0.9.0. Please use "
-            "load_sample_data(name='japan_quakes') instead.",
-            category=FutureWarning,
-            stacklevel=2,
-        )
-
     fname = which("@tut_quakes.ngdc", download="c")
-    data = pd.read_csv(fname, header=1, sep=r"\s+")
-    data.columns = [
-        "year",
-        "month",
-        "day",
-        "latitude",
-        "longitude",
-        "depth_km",
-        "magnitude",
-    ]
-
-    return data
+    return pd.read_csv(
+        fname,
+        header=1,
+        delim_whitespace=True,
+        names=[
+            "year",
+            "month",
+            "day",
+            "latitude",
+            "longitude",
+            "depth_km",
+            "magnitude",
+        ],
+    )
```

Comment:
FYI, I also made two changes when reading the dataset:

1. Use `delim_whitespace=True` instead of `sep=r"\s+"`. They're the same but I think `delim_whitespace=True` is easier to understand (just my own preference).
2. Use the `names` parameter in the `read_csv` function, instead of changing the columns later.